Programming Foundations: Algorithms/Factorial-test/fact.rb

```diff
@@ -0,0 +1,21 @@
+class Fact
+
+  def initialize(n)
+    @n = n
+  end
+
+  def fact
+    if @n == 0
+     1
+    elsif (@n < 0)
+      raise 'negatives are not acceptable'
+    else
+      result = 1
+      while @n > 0 do
+       result = result * @n
+       @n = @n - 1
+     end
+     result
+   end
```

Comment:
Be wary of indentation, it makes the code more readable.
You probably want to setup your IDE with correctly.

That makes me think of a good read for start https://github.com/rubocop-hq/ruby-style-guide